test/viewmodels/home_view_model_unit_test.dart

```diff
@@ -0,0 +1,295 @@
+          );
+        }
+
+        when(geoLocationService.getCurrentPosition()).thenAnswer(
+          (_) async => point,
+        );
+        when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+            .thenAnswer(
+          (_) async => posts,
+        );
+
+        // Check the actual posts
+        final actualPosts = await container.read(postOverviewProvider.future);
+
+        expect(actualPosts, expectedPosts);
+      },
+    );
+
+    test("New posts are exposed correctly on refresh", () async {
+      when(geoLocationService.getCurrentPosition()).thenAnswer(
+        (_) async => point,
+      );
+      when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+          .thenAnswer(
+        (_) async => [],
+      );
+
+      // Query the posts a first time
+      final postBeforeRefresh =
+          await container.read(postOverviewProvider.future);
+      expect(postBeforeRefresh, []);
+
+      // Simulate a new post being added
+      final owner = MockUserFirestore.generateUserFirestore(1)[0];
+      final postData = MockPostFirestore.generatePostData(1)
+          .map(
+            (postData) => PostData(
+              ownerId: owner.uid,
+              title: postData.title,
+              description: postData.description,
+              publicationTime: postData.publicationTime,
+              voteScore: postData.voteScore,
+            ),
+          )
+          .toList()[0];
+      final post = MockPostFirestore.createPostAt(postData, point);
+
+      when(userRepository.getUser(post.data.ownerId)).thenAnswer(
+        (_) async => owner,
+      );
+
+      final expectedPosts = [
+        PostOverview(
+          title: post.data.title,
+          description: post.data.description,
+          votes: post.data.voteScore,
+          commentNumber: 0,
+          posterUsername: owner.data.username,
+        ),
+      ];
+
+      when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+          .thenAnswer(
+        (_) async => [post],
+      );
+      when(userRepository.getUser(post.data.ownerId)).thenAnswer(
+        (_) async => owner,
+      );
+
+      // Refresh the posts
+      await container.read(postOverviewProvider.notifier).refresh();
+
+      // Check the actual posts
+      final postAfterRefresh =
+          await container.read(postOverviewProvider.future);
+
+      expect(postAfterRefresh, expectedPosts);
+    });
+
+    test("Error is exposed correctly on refresh", () async {
+      when(geoLocationService.getCurrentPosition()).thenAnswer(
+        (_) async => point,
+      );
+      when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+          .thenThrow(Exception("Error"));
+
+      // Refresh the posts
+      await container.read(postOverviewProvider.notifier).refresh();
+
+      // Check the actual posts
+      final asyncPosts = container.read(postOverviewProvider);
+
+      expect(asyncPosts, isA<AsyncError>());
```

Comment:
On this test, we should maybe test whether the message of the thrown exception is the expected one.

```suggestion
      const expectedErrorMessage = "Error while fetching posts";
      when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
          .thenThrow(Exception(expectedErrorMessage));

      // Refresh the posts
      await container.read(postOverviewProvider.notifier).refresh();

      // Check the actual posts
      final asyncPosts = container.read(postOverviewProvider);

      expect(
        asyncPosts,
        isA<AsyncError>().having(
          (error) => error.error.toString(),
          "message",
          "Exception: $expectedErrorMessage",
        ),
      );
```

Reply:
Good suggestion !
As you said, this will make sure that the right exception is thrown leading to a more robust test :+1: 

Applied in fd2e9639c0ef77cfa13fd33552841bd13b31a4ad.